src/Language/ASKEE/Exposure/Interpreter.hs

```diff
@@ -184,21 +185,11 @@ getVarValue i =
 
 bindTopLevelVar :: Ident -> Value -> Eval ()
 bindTopLevelVar i v =
-  do  mbV <- State.gets (Map.lookup i . envTopLevelVars)
-      case mbV of
-        Nothing ->
-          State.modify (\env -> env { envTopLevelVars = Map.insert i v (envTopLevelVars env) })
-        Just _ ->
-          throw ("variable " <> i <> " is already bound here")
+  State.modify (\env -> env { envTopLevelVars = Map.insert i v (envTopLevelVars env) })
 
 withLocalVar :: Ident -> Value -> Eval a -> Eval a
-withLocalVar i v thing =
-  do  localEnv    <- Reader.asks erLocalVars
-      topLevelEnv <- State.gets envTopLevelVars
-      case Map.lookup i localEnv <|> Map.lookup i topLevelEnv of
-        Nothing -> Reader.local (\er -> er { erLocalVars = Map.insert i v (erLocalVars er) }) thing
-        Just _ ->
-          throw ("variable " <> i <> " is already bound here")
+withLocalVar i v =
+  Reader.local (\er -> er { erLocalVars = Map.insert i v (erLocalVars er) })
```

Comment:
This changes the behavior of `withLocalVar` to allow local variables to shadow top-level definitions, right?

Reply:
That's right -- doesn't exactly belong with this PR logically, but it's small enough that I snuck it in to its own commit here.

Reply:
That's right -- doesn't exactly belong with this PR logically, but it's small enough that I snuck it in to its own commit here.